commit 19f00e66f8aa7ee581c6d003fd68ee9f9dee4057
tree a0cdacff1f3b06387b8c50f1c0366db236285a3d
parent 650c12c528d3e0ac69405dd35d3bc8a7228e49f2
author Stefan Richter <stefanr@s5r6.in-berlin.de> 1179766326 +0200
committer Stefan Richter <stefanr@s5r6.in-berlin.de> 1184018858 +0200

    ieee1394: raw1394: Add ioctl() for 32bit userland on 64bit kernel, amendment
    
    Pointed out by Arnd Bergmann:  PPC32 aligns this at 64bit, IA32 packs
    it.  A kernel-wide available __compat_u64 which is 4-byte aligned on
    AMD64 and IA64 would be nicer though.
    
    Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>

---
 drivers/ieee1394/raw1394.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/ieee1394/raw1394.c	2007-08-27 14:01:24.000000000 -0400
+++ b/drivers/ieee1394/raw1394.c	2007-08-27 14:02:50.000000000 -0400
@@ -2814,7 +2814,11 @@ struct raw1394_iso_packets32 {
 struct raw1394_cycle_timer32 {
         __u32 cycle_timer;
         __u64 local_time;
-} __attribute__((packed));
+}
+#if defined(CONFIG_X86_64) || defined(CONFIG_IA64)
+__attribute__((packed))
+#endif
+;
 
 #define RAW1394_IOC_ISO_RECV_PACKETS32          \
         _IOW ('#', 0x25, struct raw1394_iso_packets32)
